File: Week 9/JavaScript/reverseBinary.js
/**
 * Prompt:
 *
 * Reverse bits of a given 32 bits unsigned integer.
 *
 * https://leetcode.com/problems/reverse-bits/
 */

/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */

let reverseBits = (n) => {
  const bits = n.toString(2).padStart(32, "0").split("").reverse().join("");
  return parseInt(bits, 2);
};

/**
 * Not the greatest stats here either. Runtime of 120ms (faster than 8.42%) and memory usage of 40.5mb (less than 44.13%)
 */

File: First Job Hunt/Week 9/Java/reverseBinary.java
/**
 * Prompt:
 * 
 * Reverse bits of a given 32 bits unsigned integer.
 * 
 * https://leetcode.com/problems/reverse-bits/
 */

public class Solution {
    public int reverseBits(int n) {
        int ans = 0;
        for (int i = 0; i < 32; i++) {
            ans <<= 1;
            ans = ans | (n & 1);
            n >>= 1;
        }
        return ans;
    }
}

/**
 * Not the best results here but it gets the job done! Runtime of 3ms (faster
 * than 6.85%) and memory usage of 39mb (less than 29.84%)
 */